src/main/scala/chisel3/util/BitSet.scala

```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.util
+
+/** A Set of [[BitPat]] represents a set of bit vector with mask. */
+trait BitSet { outer =>
```

Comment:
This being an unsealed trait makes me nervous. Can we put it in the same file with `BitPat` and seal it?

Reply:
I'm still thinking should this be sealed or public. 

If we seal it, user can't extend from it, if they need to add new functions, they have to use implicit class or wrapper another function to it. There is a use case which users can extend a `BitSet` in their own class, adding more function to it, for example: `AddressSet`, some functions from that class are too specific which should not be absorbed by chisel upstream, but an `AddressSet` is still a `BitSet`, and being able to use the `BitSet` decoder API.